assembl/graphql/vote_session.py

```diff
@@ -0,0 +1,131 @@
+import graphene
+from graphene.relay import Node
+from graphene_sqlalchemy import SQLAlchemyObjectType
+from assembl import models
+import os
+
+from .types import SecureObjectType
+from .utils import abort_transaction_on_exception
+from .graphql_langstrings_helpers import (
+    LangstringsInterface,
+    update_langstrings,
+    add_langstrings_input_attrs
+)
+from .document import Document
+
+# from assembl.auth import IF_OWNED, CrudPermissions
+# from assembl.auth.util import get_permissions
+# from pyramid.httpexceptions import HTTPUnauthorized
+# from pyramid.security import Everyone
+
+
+def add_and_flush(model):
+    db = model.db
+    db.add(model)
+    db.flush()
+
+
+"""
+
+def require_permission(cls, user_id, discussion_id, permission):
+    permissions = get_permissions(user_id, discussion_id)
+    allowed = cls.user_can_cls(
+        user_id, permission, permissions)
+    if not allowed or (allowed == IF_OWNED and user_id == Everyone):
+        raise HTTPUnauthorized()
+
+def require_create_permission(cls, user_id, discussion_id):
+    require_permission(cls, user_id, discussion_id, CrudPermissions.UPDATE)
+
+def require_update_permission(cls, user_id, discussion_id):
+    require_permission(cls, user_id, discussion_id, CrudPermissions.CREATE)
+
+"""
+
+langstrings_defs = {
+    "title": {},
+    "sub_title": {},
+    "instructions_section_title": {},
+    "instructions_section_content": {},
+    "propositions_section_title": {}
+}
+
+
+class VoteSession(SecureObjectType, SQLAlchemyObjectType):
+    class Meta:
+        model = models.VoteSession
+        interfaces = (Node, LangstringsInterface(langstrings_defs))
```

Comment:
Your code here only works because you use `LangstringsInterface(langstrings_defs)` only once. If you use it a second time with different parameters, you need to have another generated interface name.
The code you wrote here seems quite complex for no added benefits in my opinion.

Reply:
I fixed what you pointed. DRY and maintanability is the big benefit, they are defined in the same way and we can be assured that they have the same behaviour without checking 15 lines of fields and columns declaration for each langstring